src/NumericalAlgorithms/SphericalHarmonics/Python/Strahlkorper.cpp

```diff
@@ -0,0 +1,44 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/SphericalHarmonics/Python/Strahlkorper.hpp"
+
+#include <array>
+#include <cstddef>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/ModalVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp"
+
+namespace py = pybind11;
+
+namespace ylm::py_bindings {
+void bind_strahlkorper(pybind11::module& m) {  // NOLINT
+  py::class_<ylm::Strahlkorper<Frame::Inertial>>(m, "Strahlkorper")
+      .def(py::init<size_t, size_t, double, std::array<double, 3>>(),
+           py::arg("l_max"), py::arg("m_max"), py::arg("radius"),
+           py::arg("center"))
+      .def(py::init<size_t, size_t, const DataVector&, std::array<double, 3>>(),
+           py::arg("l_max"), py::arg("m_max"),
+           py::arg("radius_at_collocation_points"), py::arg("center"))
+      .def(
+          py::init<size_t, size_t, const ylm::Strahlkorper<Frame::Inertial>&>(),
+          py::arg("l_max"), py::arg("m_max"), py::arg("another_strahlkorper"))
+      .def(
+          py::init<size_t, size_t, const ModalVector&, std::array<double, 3>>(),
```

Comment:
If both `DataVector` and `ModalVector` can be represented as numpy arrays, how does it know which constructor to use if a numpy array is passed? Or do you have to specify the explicit data type?

Reply:
Not sure, maybe it selects the first constructor that works. You can pass an explicit DataVector or ModalVector though.

Reply:
I know you *can* but I'm worried users won't know right away to do that and there will be some weird results by just assuming a numpy array will work and it using the wrong constructor. I don't really have a change request but something to keep aware of.